tools/github-event-processor/RULES.md

```diff
@@ -0,0 +1,638 @@
+
+  The set of owners responsible for triage for a given service may not be the same as those who should be tagged as pull request reviewers.  It should not be assumed that triage owners can be correctly inferred from the existing path/label definitions in CODEOWNERS.
+
+  Example:
+
+  ```json
+  {
+    "label" : "Event Hubs",
+    "contacts" : ["@person1"]
+  }
+  ```
+
+- **File paths**: A service label may have one or more file paths relative to the repository structure associated with it.
+
+  Example:
+
+  ```json
+  {
+    "label" : "Tables",
+    "paths" : ["/sdk/tables/Azure.Data.Tables", "/sdk/samples/some-table-sample"]
+  }
+  ```
+
+## External integrations
+
+### AI label service
+
+This is a REST API that evaluates the content of an issue and attempts to predict a set of labels that should be applied to it.  If a prediction is made with a reasonable level of confidence, the service will return exactly one service label and one category label.  If confidence thresholds were not met for both label types, no labels are returned.
+
+#### Example payload
+
+```json
+{
+  "issueNumber": 32227,
+  "title": "Move Snippet Generator to .NET 6",
+  "body": "The snippet generator tool currently targets netcoreapp 3.1 which reaches end-of-life in December, 2022. The target framework should be updated to net6.0 and the associated Update-Snippet script should be updated to reference the new version.",
+  "issueUserLogin": "person1",
+  "repositoryName": "azure-sdk-for-net",
+  "repositoryOwnerName": "Azure"
+}
+```
+
+#### Example responses
+
+_**Predictions made**_
+
+```json
+{
+  "labels": [
+    "Storage",
+    "Client"
+  ]
+}
+```
+
+_**No predictions**_
+
+```json
+{
+  "labels": []
+}
+```
+
+### Important context
+
+Today, this functionality exists as part of an Azure Function that hooks into our Event Hubs feed for GitHub events and operates independent of FabricBot and other automation.  It will need to be revised to expose an API that can be called on demand from the actions-based rules, or we will need to determine a way to more directly integrate it into the new platform.
+
+For the purposes of the rule definitions herein, it is assumed that this is a stand-alone service providing a REST API.
+
+# Issue Rules
+
+## Initial issue triage
+
+### Trigger
+
+- Issue created
+
+### Criteria
+
+- Issue has no labels
+- Issue has no assignee
+
+### Actions
+
+- Evaluate the user that created the issue:
+
+    ```text
+    IF creator is NOT an Azure SDK team owner
+      AND is NOT a member of the Azure organization
+      AND is does NOT have a collaborator association
+      AND does NOT have write permission
+      AND does NOT have admin permission:
+
+          - Add "customer-reported" label
+          - Add "question" label
+    ```
+
+- Query AI label service for suggestions:
```

Comment:
I believe you have ported Jesse's rule requirements here but we should make sure to at least call out anything that currently doesn't work in the tool as we don't want to confuse what we support now from what we want to support eventually. We should probably file issues for any anything we don't support intially but plan to.

Reply:
Agreed. I've put in what we have today and show where we're like to get to and why it's not there yet. This will be updated with the next commit.